---
title: "SureTypeSCR -- library for (single cell) SNP array data processing"
author: "Ivan Vogel, Lishan Cai"
date: "`r format(Sys.time(), '%B %d, %Y')`"
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{SureTypeSCR overview}
  %\VignetteEncoding{UTF-8}
output:
  BiocStyle::html_document:
    highlight: pygments
    number_sections: yes
    theme: united
    toc: yes
---

# Introduction

SureTypeSCR is an R package for QC and rapid genotyping of single cell SNP array data. It encapsulates previously developed library SureTypeSC (Vogel et al., 2019, PMID: 31116387). The core consists of a two layered machine learning method that assigns a quality score to each SNP in an array. On top of that, SureTypeSCR implements various QC strategies to examine the data. SureTypeSCR is extensively using packages from the tidyverse collection thereby allowing to apply modern data science approach to query the data.

This tutorial gives basic introduction to the data analysis with SureTypeSCR. The data used in this tutorial is publicly available (Johnson et al.,2010, PMID: 20100701) and subset of it (sperm data) was transformed into an R data package for purpose of this demonstration.

# Basic concepts

## Installation of the program and the data

Both, SureTypeSCR and johnsonspermdata are installed from github repository using devtools. The dependencies should be installed automatically.

```{r install,echo=TRUE,results="show",include=TRUE,cache=FALSE}
#require(nothing, quietly = TRUE)

#library(devtools)

##install  SureTypeSCR  from  github
#devtools::install_github("Meiomap/SureTypeSCR")

##install  data  package  with  sperm  data (compiled  from  GSE19247)
#devtools::install_github("Meiomap/johnsonspermdata")
```

When loaded for the first time, SureTypeSCR will create a python virtual environment and install all required python libraries via python package installer (pip).

```{r init, echo=FALSE, warning=FALSE, results='hide'}
library(SureTypeSCR)
library(tidyverse)
library(johnsonspermdata)
library(magrittr)
library(ggrepel)
# load  metadata  and  samplesheet  location
data(metadata ,samplesheet)
```

```{r setup, include=FALSE}
require('knitr')
##setting working directory
knitr::opts_knit$set(root.dir=system.file("data",package="johnsonspermdata"))



```

Required manifest and cluster file are provided with the package.

```{r load,echo=TRUE,results="show",include=TRUE,echo=TRUE}
manifest=system.file("files/HumanCytoSNP-12v2_H.bpm",package="SureTypeSCR")
cluster=system.file("files/HumanCytoSNP-12v2_H.egt",package="SureTypeSCR")
df = scbasic(samplesheet=samplesheet , bpm=manifest , egt=cluster)
```

We filter out markers that don't contain genotypes (intensity-only SNPs) and view the structure of the data frame.

```{r filter,echo=TRUE,results="show",include=TRUE}
#filtering  out  intensity -only  SNPs
df %<>% dplyr::filter(Chr!=0 & !str_detect(Name ,"cnv"))
head(df)

```

Call rates can be calculated per group basis - here per individual (sample) and genotype.

```{r callrates,echo=TRUE,results="show",include=TRUE}
df %>%
  group_by(individual ,gtype) %>%
  callrate()

```

Principal component analysis determines relationships between the individuals by reducing n-dimensional representation of SNP array data (samples x number_of_SNPs) into few (here two) dimensions that maximize the variation.

```{r pca,echo=TRUE,results="show",include=TRUE}
df %>%
  plot_pca(features="gtype", metadata=metadata ,by_chrom=FALSE)
```

Now we perform MA transformation and visualise:

```{r  MAtransform,echo=TRUE,results="show",include=TRUE}
df %>%
  calculate_ma() %>%
  plot_ma(n=0.1)

```

The classification of the single cell genotypes is now performed via pretrained classifier (Random Forest) that is distributed within the package. A second layer of the algorithm (Gaussian Discriminant Analysis) is executed directly on the data. The classification is performed per sample basis.

```{r classification,echo=TRUE,results="show",include=TRUE}
clf=system.file("files/rf.clf",package="SureTypeSCR")
df_model = df %>%
  calculate_ma() %>%
  group_by(individual) %>%
  nest() %>%
  mutate(model=map(data , function(df) suretype_model(df,individual , clf)))
```

Now we visualise the MA plot again - this time for the filtered data to explore the effect of the RF-GDA algorithm.

```{r After_classification,echo=TRUE,results="show",include=TRUE}
df_model_unnested = df_model %>%
  unnest(c(data ,model))

df_model_unnested %>%
  set_threshold(rfgda_score,threshold =0.5) %>%
  plot_ma()
```
