---
title: "SureTypeSCR -- R package for (single cell) SNP array data"
author: "Ivan Vogel, Lishan Cai"
date: "`r format(Sys.time(), '%B %d, %Y')`"
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{SureTypeSCR overview}
  %\VignetteEncoding{UTF-8}
output:
  BiocStyle::html_document:
    highlight: pygments
    number_sections: yes
    theme: united
    toc: yes
---

# Introduction

SureTypeSCR is an R package for QC and rapid genotyping of single cell SNP array data. It encapsulates previously developed library SureTypeSC (Vogel et al., 2019, PMID: 31116387). The core consists of a two layered machine learning method that assigns a quality score to each SNP in an array. On top of that, SureTypeSCR implements various QC strategies to examine the data. SureTypeSCR is extensively using packages from the tidyverse collection thereby allowing to apply modern data science approach to query the data.

This tutorial gives basic introduction to the data analysis with SureTypeSCR. The data used in this tutorial is publicly available (Johnson et al.,2010, PMID: 20100701) and subset of it (sperm data) was transformed into an R data package for purpose of this demonstration.

# Basic concepts

## Installation of the program and the data and initialization

Both, **SureTypeSCR** and **johnsonspermdata** can be installed from github repository using devtools. The dependencies should be installed automatically.

```{r install, warning=FALSE, cache=FALSE, results="hide",error=FALSE,message=FALSE}
library(devtools)

##install  SureTypeSCR  from  github
devtools::install_github("Meiomap/SureTypeSCR")

##install  data  package  with  sperm  data (compiled  from  GSE19247)
devtools::install_github("Meiomap/johnsonspermdata")

```

When loaded for the first time, SureTypeSCR will create a python virtual environment and install all required python libraries via python package installer (pip).

```{r setup, include=FALSE}
require('knitr')
##setting working directory
knitr::opts_knit$set(root.dir=system.file("data",package="johnsonspermdata"))

```

```{r init, echo=TRUE, warning=FALSE, results='hide',error=FALSE,message=FALSE}
library(SureTypeSCR)
library(johnsonspermdata)
#library(tidyverse)
#library(magrittr)
#library(ggrepel)
# load  metadata  and  samplesheet  location
data(metadata ,samplesheet)
```

## Data loading and feature extraction

Metadata that is required for feature extraction and creation of the basic data frame is deployed with the package. Note that these metadata (except for the samplesheet) are array specific and differ accross Illumina products. The program lists the sample files as being analyzed and gives dimensions of the final data frame to the output (before and after applying internal indexing).

```{r load,echo=TRUE,results="show",include=TRUE,echo=TRUE,cache=TRUE}
manifest=system.file("files/HumanCytoSNP-12v2_H.bpm",package="SureTypeSCR")
cluster=system.file("files/HumanCytoSNP-12v2_H.egt",package="SureTypeSCR")
df = scbasic(samplesheet=samplesheet , bpm=manifest , egt=cluster)
```

## Filtering

After loading the basic dataframe **df** we filter out markers that don't contain genotypes (intensity-only SNPs) and view the structure of the data frame.

```{r filter,echo=TRUE,results="show",include=TRUE}
#filtering  out  intensity -only  SNPs
df %<>% dplyr::filter(Chr!=0 & !str_detect(Name ,"cnv"))
head(df)

```

## Call rates

Call rates can be calculated per group basis - here per individual (sample) and genotype.

```{r callrates,echo=TRUE,results="show",include=TRUE}
df %>%
  group_by(individual ,gtype) %>%
  callrate() %>%
  pivot_wider(values_from=Callrate,names_from=gtype)

```

## Principal component analysis

Principal component analysis determines relationships between the individuals by reducing n-dimensional representation of SNP array data (samples x number_of_SNPs) into few (here two) dimensions that maximize the variation.

```{r pca,echo=TRUE,results="show",include=TRUE,cache=TRUE}
df %>%
  plot_pca(features="gtype", metadata=metadata ,by_chrom=FALSE)
```

## MA transformation

MA transformation displays intensities as logarithmic average (A) and logarithmic difference (M). Roughly speaking, M\~0 corresponds to heterozygous genotypes, M\<0 corresponds to homozygous B and M\>0 corresponds to homozygous A.

```{r  MAtransform,echo=TRUE,results="show",include=TRUE,cache=TRUE}
df %>%
  calculate_ma() %>%
  plot_ma(n=0.1)

```

## Single cell genotype classification

The classification of the single cell genotypes is now performed via pretrained classifier (Random Forest) that is distributed within the package. A second layer of the algorithm (Gaussian Discriminant Analysis) is executed directly on the data. The model is created per sample basis.

```{r classification,echo=TRUE,results="show",include=TRUE,cache=TRUE}
clf=system.file("files/rf.clf",package="SureTypeSCR")
df_model = df %>%
  calculate_ma() %>%
  group_by(individual) %>%
  nest() %>%
  mutate(model=map(data , function(df) suretype_model(df,individual , clf)))
head(df_model)
```

Now we visualise the MA plot again - this time for the filtered data to explore the effect of the RF-GDA algorithm. The algorithm has effectively filtered out the heterozygous genotypes (m \~ 0)

```{r After_classification,echo=TRUE,results="show",include=TRUE,cache=TRUE}
df_model_unnested = df_model %>%
  unnest(c(data ,model))
head(df_model_unnested)
df_model_unnested %>%
  set_threshold('rfgda_score',threshold =0.5) %>%
  plot_ma(n=0.1)
```

### Conditional processing

An optional parameter of function **suretype_model(.)** is **.sclist** that controls samples that will be processed using RF-GDA algorithm (by default, all samples in the dataset are analysed). Following example demonstrates situation when we only want to analyze subset of samples as single cells - this is particularly useful in case there are also bulk DNA samples (i.e. parental genotypes) in the dataset.

```{r Sample_list,echo=TRUE,results="show",include=TRUE}
df %>% 
  select(individual) %>%
  distinct()

```

We choose sample **gsm477532** and **gsm477533** for classification.

```{r Subset_classification,echo=TRUE,results="show",include=TRUE,cache=TRUE}
sample_subset=list('gsm477532','gsm477533')

df_model = df %>%
  calculate_ma() %>%
  group_by(individual) %>%
  nest() %>%
  mutate(model=map(data , function(df) suretype_model(df,individual,clf,.sclist=sample_subset)))
```

Mean and standard deviation of the score per sample clearly shows that only the selected samples were process while all the other samples were assigned score 1 for all genotypes.

```{r  Subset_score_distribution,echo=TRUE,results="show",include=TRUE,cache=TRUE}
df_model %>%
  unnest(c(data ,model)) %>%
  group_by(individual)  %>%
  summarize(mean = mean(rfgda_score,na.rm=TRUE),
    sd = sd(rfgda_score,na.rm=TRUE))

```

## Details

For detailed description of available functions and parameters please see the reference manual (vignette). 
