\nonstopmode{}
\documentclass[a4paper]{book}
\usepackage[times,inconsolata,hyper]{Rd}
\usepackage{makeidx}
\usepackage[utf8,latin1]{inputenc}
% \usepackage{graphicx} % @USE GRAPHICX@
\makeindex{}
\begin{document}
\chapter*{}
\begin{center}
{\textbf{\huge Package `SureTypeSCR'}}
\par\bigskip{\large \today}
\end{center}
\begin{description}
\raggedright{}
\item[Title]\AsIs{Interface to python based package SureTypeSC via reticulate}
\item[Description]\AsIs{SureTypeSCR is the implementation of algorithm for regenotyping of single cell data coming from Illumina BeadArrays without genome studio.}
\item[Version]\AsIs{0.99.0}
\item[Author]\AsIs{Ivan Vogel, Lishan Cai}
\item[Suggests]\AsIs{testthat}
\item[Depends]\AsIs{R (>= 3.5.0), reticulate, knitr, BiocStyle}
\item[Maintainer]\AsIs{Lishan Cai }\email{Lishan@sund.dk.com}\AsIs{}
\item[License]\AsIs{Artistic-2.0}
\item[Imports]\AsIs{reticulate, knitr, BiocStyle}
\item[biocViews]\AsIs{Software, GenotypingArray,SingleCell}
\item[VignetteBuilder]\AsIs{knitr}
\item[SystemRequirements]\AsIs{python (>= 2.7), sklearn, numpy, pandas,
SureTypeSC, IlluminaBeadArrayFiles}
\item[NeedsCompilation]\AsIs{no}
\end{description}
\Rdcontents{\R{} topics documented:}
\inputencoding{utf8}
\HeaderA{allele\_freq}{The frequency function is to calculate the allele frequency over all the samples}{allele.Rul.freq}
%
\begin{Description}\relax
The frequency function is to calculate the allele frequency over all the samples
\end{Description}
%
\begin{Usage}
\begin{verbatim}
allele_freq(df,th=0)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] the pandas dataframe from GenomeStudio or scbasic function
\item[\code{th}] the threshold
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The frequency values of all samples
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)

# The Random Forest classifier
call <- allele_freq(df,th=0.2) 




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{apply\_thresh}{To apply threshold over all samples on GenCall score}{apply.Rul.thresh}
%
\begin{Description}\relax
To apply threshold over all samples on GenCall score
\end{Description}
%
\begin{Usage}
\begin{verbatim}
apply_thresh(df,th)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] the Data object from create\_from\_frame function
\item[\code{th}] the threshold
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Data object only with applied threshold
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)

df <- create_from_frame(df)

df <- apply_thresh(df,0.01)




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{calculate\_ma}{To calculate m and a features}{calculate.Rul.ma}
%
\begin{Description}\relax
To calculate m and a features
\end{Description}
%
\begin{Usage}
\begin{verbatim}
calculate_ma(df)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] the Data object from create\_from\_frame function
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Data object with adding m and a features
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)

df <- create_from_frame(df)

dfs <- calculate_ma(df)




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{callrate}{The callrate function is to calculate the allele frequency over all the samples}{callrate}
%
\begin{Description}\relax
The callrate function is to calculate the allele frequency over all the samples
\end{Description}
%
\begin{Usage}
\begin{verbatim}
callrate(df,th=0)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] the pandas dataframe from GenomeStudio or scbasic function
\item[\code{th}] the threshold
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The callrate values of all samples
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)

# The Random Forest classifier
call <- callrate(df,th=0.2) 




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{callrate\_chr}{The callrate function is to calculate the allele frequency over all the samples of one specific chromosome}{callrate.Rul.chr}
%
\begin{Description}\relax
The callrate function is to calculate the allele frequency over all the samples of one specific chromosome
\end{Description}
%
\begin{Usage}
\begin{verbatim}
callrate_chr(df,chr_name,th=0)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] the pandas dataframe from GenomeStudio or scbasic function
\item[\code{chr\_name}] the name of the selected chromsome
\item[\code{th}] the threshold
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The callrate values of all samples of one specific chromosome
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)

# The Random Forest classifier
call <- callrate_chr(df,'1',th=0.2) 




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{create\_from\_frame}{convert pandas dataframe to Data object}{create.Rul.from.Rul.frame}
%
\begin{Description}\relax
Convert pandas dataframe to Data object and rearrange the index level
\end{Description}
%
\begin{Usage}
\begin{verbatim}
create_from_frame(df)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] genotyping pandas dataframe from scbasic function

\end{ldescription}
\end{Arguments}
%
\begin{Value}
Data object with index rearrangement (multi-leve index)

\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

# get genotyping data from gtc files and meta file
df <- scbasic(manifest_path,cluster_path,samplesheet)


# create Data object and rearrange the index
df <- create_from_frame(df)


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{locus\_cluster}{To do intensity aggregation at a specific locus}{locus.Rul.cluster}
%
\begin{Description}\relax
To do intensity aggregation at a specific locus
\end{Description}
%
\begin{Usage}
\begin{verbatim}
locus_cluster(df,locus)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] the pandas dataframe from GenomeStudio or scbasic function
\item[\code{locus}] the name of the locus
\end{ldescription}
\end{Arguments}
%
\begin{Value}
the intensity of one locus
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)

# The Random Forest classifier
call <- locus_cluster(df,'rs3128117') 




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{locus\_ma}{To do m and a aggregation at a specific locus}{locus.Rul.ma}
%
\begin{Description}\relax
To do m and a aggregation at a specific locus
\end{Description}
%
\begin{Usage}
\begin{verbatim}
locus_ma(df,locus)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] the pandas dataframe from GenomeStudio or scbasic function
\item[\code{locus}] the name of the locus
\end{ldescription}
\end{Arguments}
%
\begin{Value}
the m and a of one locus
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)

# The Random Forest classifier
call <- locus_ma(df,'rs3128117') 




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{pca\_chr}{To apply principle component annalysis on frequency dataframe of samples of one chromosome}{pca.Rul.chr}
%
\begin{Description}\relax
To apply principle component annalysis on frequency dataframe of samples
\end{Description}
%
\begin{Usage}
\begin{verbatim}
pca_chr(df,chr_name,th=0,n=2)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] the pandas dataframe from GenomeStudio or scbasic function
\item[\code{chr\_name}] the name of the specific chromosome
\item[\code{th}] the threshold
\item[\code{n}] n is the number of components
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Component values
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)

# The Random Forest classifier
call <- pca_chr(df,'X') 




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{pca\_samples}{To apply principle component annalysis on frequency dataframe of samples}{pca.Rul.samples}
%
\begin{Description}\relax
To apply principle component annalysis on frequency dataframe of samples
\end{Description}
%
\begin{Usage}
\begin{verbatim}
pca_samples(df,th=0)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] the pandas dataframe from GenomeStudio or scbasic function
\item[\code{th}] the threshold
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Component values
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)

# The Random Forest classifier
call <- pca_samples(df,th=0.2) 




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{restrict\_chrom}{To choose certain chromosomes with Data object}{restrict.Rul.chrom}
%
\begin{Description}\relax
To choose certain chromosomes with Data object
\end{Description}
%
\begin{Usage}
\begin{verbatim}
restrict_chrom(df,chrom)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] the Data object from create\_from\_frame function
\item[\code{chrom}] the list of selected chromosomes
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Data object only with certain chromosomes
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)

df <- create_from_frame(df)

df <- restrict_chrom(df,c('1','2'))




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{sample\_ma}{To do m and a aggregation at a specific chromosome of a specific sample}{sample.Rul.ma}
%
\begin{Description}\relax
To do m and a at a specific chromosome of a specific sample
\end{Description}
%
\begin{Usage}
\begin{verbatim}
sample_ma(df,sample_name,chr_name)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] the pandas dataframe from GenomeStudio or scbasic function
\item[\code{sample\_name}] the name of the sample
\item[\code{chr\_name}] the name of the chromosome
\end{ldescription}
\end{Arguments}
%
\begin{Value}
the m and a a of specific chromosome of a specific sample
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)

# The Random Forest classifier
call <- sample_ma(df,'Kit4_4mos_SC21','1') 



\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{scbasic}{Function to process raw gtc data and meta data without genomestudio}{scbasic}
%
\begin{Description}\relax
Function to process raw gtc data and meta data without genomestudio
\end{Description}
%
\begin{Usage}
\begin{verbatim}
scbasic(bpm,egt,samplesheet)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}

\item[\code{bpm}] a pathname to manifest file

\item[\code{egt}] a pathname to cluster file

\item[\code{samplesheet}] a pathname to samplesheet file

\end{ldescription}
\end{Arguments}
%
\begin{Value}
pandas data frame of genotyping data 
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

# get genotyping data from gtc files and meta file
df <- scbasic(manifest_path,cluster_path,samplesheet)

#/Users/apple/anaconda3/envs/gtc2/lib/python2.7/site-packages/sklearn/ensemble/weight_boosting.py:29: 
#DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. 
#It will be removed in a future NumPy release.
  #from numpy.core.umath_tests import inner1d
#Reading cluster file
#Reading sample file
#Number of samples: 2
#Reading manifest file
#Initializing genotype data
#Generating
#9968648019_R06C01
#9968648019_R06C02
#Finish parsing



\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{scEls}{mediate access to python modules}{scEls}
%
\begin{Description}\relax
mediate access to python modules
\end{Description}
%
\begin{Usage}
\begin{verbatim}
scEls()
\end{verbatim}
\end{Usage}
%
\begin{Value}
list of (S3) "python.builtin.module"
\end{Value}
%
\begin{Note}\relax
Returns a list with elements sc (SureTypeSC), pd (pandas)each
referring to python modules.
\end{Note}
%
\begin{Examples}
\begin{ExampleCode}
els = scEls()

els

##$sc
##Module(SureTypeSC)

##$pd
##Module(pandas) 
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{scload}{Load Random Forest classifier or Gaussian Discrinimate Analysis}{scload}
%
\begin{Description}\relax
Load Random Forest classifier or Gaussian Discrinimate Analysis
\end{Description}
%
\begin{Usage}
\begin{verbatim}
scload(filename)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{filename}] a pathname to an classifier

\end{ldescription}
\end{Arguments}
%
\begin{Value}
instance of a classifier
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

clf_rf_path = system.file('files/clf_30trees_7228_ratio1_lightweight.clf',package='SureTypeSCR')
clf_gda_path = system.file('files/clf_30trees_7228_ratio1_lightweight.clf',package='SureTypeSCR')

# The Random Forest classifier
clf_rf <- scload(clf_rf_path) 

# The Gaussian Discriminate Analysis classifier
clf_gda <- scload(clf_gda_path) 



\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{scpredict}{Predictions from Random Forest classifier or Gaussian Discriminant Analysis}{scpredict}
%
\begin{Description}\relax
Predictions from Random Forest classifier or Gaussian Discriminant Analysis

\end{Description}
%
\begin{Usage}
\begin{verbatim}

scpredict(clf,test,clftype='rf')

\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{clf}] classifier load by using scload
\item[\code{test}] Data object including m and a feature
\item[\code{clftype}] The type of classifier 
(rf: Random Forest; 

gda: Gaussian Discriminant Analysis; 

rf-gda: the cascade of Random Forest and Gaussian Discriminant Analysis)

\end{ldescription}
\end{Arguments}
%
\begin{Value}
The prediction Data object.  

The predicted items might include:

rf\_ratio:1\_pred: Random Forest prediction (binary) 

rf\_ratio:1\_prob: Random Forest Score for the positive class

gda\_ratio:1\_prob: Gaussian Discriminant Analysis score for the positive class 

gda\_ratio:1\_pred: Gaussian Disciminant Analysis prediction (binary)

rf-gda\_ratio:1\_prob: combined 2-layer RF and GDA - probability score for the positive class

rf-gda\_ratio:1\_pred: binary prediction of RF-GDA 

\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')
clf_rf_path = system.file('files/clf_30trees_7228_ratio1_lightweight.clf',package='SureTypeSCR')
clf_gda_path = system.file('files/clf_30trees_7228_ratio1_lightweight.clf',package='SureTypeSCR')



# The Random Forest classifier
clf_rf <- scload(clf_rf_path) 

# The Gaussian Disciminant Analysis
clf_gda <- scload(clf_gda_path) 

# get genotyping data from gtc files and meta file
df <- scbasic(manifest_path,cluster_path,samplesheet)
 
# create Data object and rearrange the index 
dfs <- create_from_frame(df) 

# extract the chromosomes 1 and 2
dfs <- restrict_chrom(dfs,c('1','2')) 

# mask the Gencall score lower than 0.01
dfs <- apply_thresh(dfs,0.01) 

# calculate the m and a feature
dfs <- calculate_ma(dfs) 

# prediction by Random Forest
result_rf <- scpredict(clf_rf,dfs,clftype='rf')

# prediction by Guassian Discriminate Analysis
result_gda <- scpredict(clf_gda,dfs,clftype='gda')


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{scsave}{Save the predictions from different classifiers}{scsave}
%
\begin{Description}\relax
Save the predictions from different classifiers
\end{Description}
%
\begin{Usage}
\begin{verbatim}

# save different mdoes based on the full prediction table 
scsave(result,filename,header=TRUE,clftype='rf',threshold=0.15,all=FALSE) 

\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{result}] The predicted result

\item[\code{filename}] The path where the result will be saved

\item[\code{header}] the index

\item[\code{clftype}] classifier type

\item[\code{threshold}] the threshold of gencall score

\item[\code{all}] if the users want to save the full table or not

\end{ldescription}
\end{Arguments}
%
\begin{Value}
txt file of the results
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')
clf_rf_path = system.file('files/clf_30trees_7228_ratio1_lightweight.clf',package='SureTypeSCR')
clf_gda_path = system.file('files/clf_30trees_7228_ratio1_lightweight.clf',package='SureTypeSCR')


# The Random Forest classifier
clf_rf = scload(clf_rf_path) 

# The Gaussian Disciminant Analysis
clf_gda = scload(clf_gda_path) 

# get genotyping data from gtc files and meta file
df <- scbasic(manifest_path,cluster_path,samplesheet)  

# create Data object and rearrange the index
dfs <- create_from_frame(df) 


#original shape (294602, 15)
#shape after operation (294602, 12)



# extract the chromosomes 1 and 2
dfs <- restrict_chrom(dfs,c('1','2')) 

# mask the Gencall score lower than 0.01
dfs <- apply_thresh(dfs,0.01) 

# calculate the m and a feature
dfs <- calculate_ma(dfs) 

# prediction by Random Forest
result_rf <- scpredict(clf_rf,dfs,clftype='rf')

# prediction by Guassian Discriminate Analysis
result_gda <- scpredict(clf_gda,dfs,clftype='gda')

# Train the rf-gda classifier
trainer <- scTrain(result_gda,clfname='gda') 

# The prediction from the cascade of Random Forest and Gaussian Discriminate Analysis
result_end <- scpredict(trainer,result_gda,clftype='rf-gda') 

# Save the complete prediction table
scsave(result_end,'fulltable.txt',clftype='rf',header=TRUE,threshold=0.15,all=TRUE)

# recall mode
scsave(result_end,'fulltable.txt',clftype='rf',threshold=0.15,header=TRUE,all=FALSE)
 


\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{scTrain}{Train Gaussian Discriminate Analysis by using the output of predicitons of Random Forest}{scTrain}
%
\begin{Description}\relax
Train Gaussian Discriminate Analysis by using the output of predicitons of Random Forest
\end{Description}
%
\begin{Usage}
\begin{verbatim}
scTrain(trainingdata, clfname='gda')
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{trainingdata}] Data object, the training data of scTrain
\item[\code{clfname}] The classifier type

\end{ldescription}
\end{Arguments}
%
\begin{Value}
instance of a classifier
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')
clf_rf_path = system.file('files/clf_30trees_7228_ratio1_lightweight.clf',package='SureTypeSCR')
clf_gda_path = system.file('files/clf_30trees_7228_ratio1_lightweight.clf',package='SureTypeSCR')



# The Random Forest classifier
clf_rf <- scload(clf_rf_path) 

# The Gaussian Disciminant Analysis
clf_gda <- scload(clf_gda_path) 

 # get genotyping data from gtc files and meta file
df <- scbasic(manifest_path,cluster_path,samplesheet)
 
# create Data object and rearrange the index
dfs <- create_from_frame(df) 

# extract the chromosomes 1 and 2
dfs <- restrict_chrom(dfs,c('1','2')) 

# mask the Gencall score lower than 0.01
dfs <- apply_thresh(dfs,0.01) 

# calculate the m and a feature
dfs <- calculate_ma(dfs) 

# prediction by Random Forest
result_rf <- scpredict(clf_rf,dfs,clftype='rf')

# prediction by Guassian Discriminate Analysis
result_gda <- scpredict(clf_gda,dfs,clftype='gda')

# Train the rf-gda classifier
trainer <- scTrain(result_rf,clfname='gda')

# The prediction from the cascade of Random Forest and Gaussian Discriminate Analysis
result_end <- scpredict(trainer,result_gda,clftype='rf-gda') 




 
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{sc\_allele\_freq}{This function is to calculate the allele frequency over all the samples}{sc.Rul.allele.Rul.freq}
%
\begin{Description}\relax
The callrate function is to calculate the allele frequency over all the samples
\end{Description}
%
\begin{Usage}
\begin{verbatim}
sc_allele_freq(df,alg,threshold)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] Data object
\item[\code{alg}] algorithm
\item[\code{threshold}] the threshold
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The callrate values of all samples 
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)


# create Data object and rearrange the index
df <- create_from_frame(df)

# The Random Forest classifier
call <- sc_allele_freq(df,'score',0.2) 




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{sc\_callrate}{The callrate function is to calculate the allele frequency over all the samples}{sc.Rul.callrate}
%
\begin{Description}\relax
The callrate function is to calculate the allele frequency rate over all the samples
\end{Description}
%
\begin{Usage}
\begin{verbatim}
sc_callrate(df,alg,threshold)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] Data object
\item[\code{alg}] algorithm
\item[\code{threshold}] the threshold
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The callrate values of all samples 
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)


# create Data object and rearrange the index
df <- create_from_frame(df)

# The Random Forest classifier
call <- sc_callrate(df,'score',0.2) 




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{sc\_callrate\_chr}{The callrate function is to calculate the allele frequency of all the samples of one chromosome}{sc.Rul.callrate.Rul.chr}
%
\begin{Description}\relax
The callrate function is to calculate the allele frequency rate over all the samples
\end{Description}
%
\begin{Usage}
\begin{verbatim}
sc_callrate_chr(df,alg,threshold,chrr)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] Data object
\item[\code{alg}] algorithm
\item[\code{threshold}] the threshold
\item[\code{chrr}] the name of chromosome
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The callrate values of all samples of one chromosome
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)


# create Data object and rearrange the index
df <- create_from_frame(df)

# The Random Forest classifier
call <- sc_callrate_chr(df,'score',0.2,'21') 




\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{sc\_chr\_freq}{This function is to calculate the allele frequency over all the samples of one chromosome}{sc.Rul.chr.Rul.freq}
%
\begin{Description}\relax
The callrate function is to calculate the allele frequency over all the samples of one chromosome
\end{Description}
%
\begin{Usage}
\begin{verbatim}
sc_chr_freq(df,alg,threshold,chrr)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{df}] Data object
\item[\code{alg}] algorithm
\item[\code{threshold}] the threshold
\item[\code{chrr}] the name of chromosome
\end{ldescription}
\end{Arguments}
%
\begin{Value}
The callrate values of all samples 
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}

# parsing file from gtc raw files

gtc_path = system.file("files/GTCs",package='SureTypeSCR')
cluster_path = system.file('files/HumanKaryomap-12v1_A.egt',package='SureTypeSCR')
manifest_path = system.file('files/HumanKaryomap-12v1_A.bpm',package='SureTypeSCR')
samplesheet = system.file('files/Samplesheetr.csv',package='SureTypeSCR')

df <- scbasic(manifest_path,cluster_path,samplesheet)


# create Data object and rearrange the index
df <- create_from_frame(df)

# The Random Forest classifier
call <- sc_chr_freq(df,'score',0.2,'21') 




\end{ExampleCode}
\end{Examples}
\printindex{}
\end{document}
